Python/optimizer.c

```diff
@@ -1069,15 +1091,19 @@ prepare_for_execution(_PyUOpInstruction *buffer, int length)
                 exit_op = _HANDLE_PENDING_AND_DEOPT;
             }
             int32_t jump_target = target;
-            if (is_for_iter_test[opcode]) {
-                /* Target the POP_TOP immediately after the END_FOR,
-                 * leaving only the iterator on the stack. */
-                int extended_arg = inst->oparg > 255;
-                int32_t next_inst = target + 1 + INLINE_CACHE_ENTRIES_FOR_ITER + extended_arg;
-                jump_target = next_inst + inst->oparg + 1;
+            bool unique_target = false;
```

Comment:
Why have you removed the `if (is_for_iter_test[opcode]) {` handling?

Reply:
I do the jump target calculation in the tracer recorder now. Not when we insert the exit traces.